### ISSUE TITLE
grid.addColumn() throws exception when Container was already set

### ISSUE THREAD

**Issue by Hacker-CB** [viritin/viritin#223]
I need to add column with actions to my grid which will be rendered using [gridactionrenderer-add-on](https://vaadin.com/directory#!addon/gridactionrenderer-add-on)

Currently it is imposible to add additional column to MGrid.
It is failed with "Property ID 'actions' does not exists in the container"
```
@Entity
class MyEntity {
    public String name;
}

MGrid <MyEntity> grid = new MGrid<>(MyEntity.class)
Column actionsColumn = grid.addColumn("actions");
```

Also I have tried next way, but failed with UnsupportedOperationException:
```
MGrid <MyEntity> grid = new MGrid<>(MyEntity.class)
grid.getContainerDataSource().addContainerProperty("actions",String.class,"");
```

Of cause, I can add extra "actions" field to my entity class, but it is not good idea because this field will be used only in grid.
So, is there any way to add extra column to grid without modifying POJO "MyEntity" class?

**Comment by Hacker-CB** [comment]
I have also tried next but also got exception "Property ID 'actions' does not exists in the container":
```
grid.withGeneratedColumn("actions", new PropertyValueGenerator<String>() {
    @Override
    public String getValue(Item item, Object itemId, Object propertyId) {
        return "1,2,3";
    }

    @Override
    public Class<String> getType() {
        return String.class;
    }
});
Column actionsColumn = grid.getColumn("actions");
```

**Comment by Hacker-CB** [comment]
It works with the following example:
```
MGrid <MyEntity> grid = new MGrid<>(MyEntity.class)
grid.withGeneratedColumn("actions", bean -> "my value");
List<Column> columns = grid.getColumns();
List<String> columnNames = new ArrayList<>();
columns.forEach(column -> columnNames.add(column.getPropertyId().toString()));
columnNames.add("actions");
grid.withProperties(columnNames.toArray(new String[columnNames.size()]));
```
So, seems that it is bug in withGeneratedColumn() call. For my opinion it should add property to container by itself, without additional withProperties() call

**Comment by sischi** [comment]
this issue seems to be related to #219

**Comment by mstahv** [comment]
I closed the related #219, but there might still be something to be done for this. This might now work if you first add the column using withGeneratedColumn and then define the renderer for the new column. I'm not big fan of renderer concept in general, but I could try this out with a fresh test project.

**Comment by mstahv** [comment]
Seems to work pretty well with GridActionrenderer like this:

        MGrid<MyPojo> g = new MGrid<>(MyPojo.class);
        
        g.setRows(Arrays.asList(new MyPojo("Nolla"), new MyPojo("Yks"), new MyPojo("Kaks")));
        
        String[] a = new String[] {"1", "1,2", "-1"};
        g.withGeneratedColumn("actions", p -> a[p.getId()%a.length]);
        
        
        List<GridAction> actions = new ArrayList<GridAction>();
        actions.add(new GridAction(FontAwesome.USER, "user"));
        actions.add(new GridAction(FontAwesome.GEAR, "settings"));
        actions.add(new GridDownloadAction(FontAwesome.TICKET, "ticket"));
        
         GridActionRenderer actionRenderer = new GridActionRenderer(actions);
        actionRenderer.addActionClickListener(new GridActionRenderer.GridActionClickListener() {
            @Override
            public void click(GridActionRenderer.GridActionClickEvent event) {
                Notification.show("It works!?");
            }
        });
        
        g.getColumn("actions").setRenderer(actionRenderer);
